core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -75,6 +76,8 @@ class BrokerServer(
   // Get raftManager from SharedServer. It will be initialized during startup.
   def raftManager: KafkaRaftManager[ApiMessageAndVersion] = sharedServer.raftManager
 
+  Server.maybeRegisterLinuxMetrics(config, time, logger.underlying)
```

Comment:
This is not a new change. The metrics are currently registered in `KafkaBroker.scala`. This just moves where they are registered. Here is the existing registration code:

```
private val linuxIoMetricsCollector = new LinuxIoMetricsCollector("/proc", Time.SYSTEM, logger.underlying)

if (linuxIoMetricsCollector.usable()) {
  metricsGroup.newGauge("linux-disk-read-bytes", () => linuxIoMetricsCollector.readBytes())
  metricsGroup.newGauge("linux-disk-write-bytes", () => linuxIoMetricsCollector.writeBytes())
}```

These metrics are described in KIP-551, which you can read here: https://cwiki.apache.org/confluence/x/sotSC

Reply:
ok. I asked this question because I didn't see any change to `KafkaBroker.scala` in this PR and assumed that we are adding new metrics. I apologize if I am missing something very obvious but I am confused here :(. Shouldn't we remove the registration from KafkaBroker since we are now registering them here?

Reply:
Yes, we should remove the registration from KafkaBroker. I'll do that now.